src/MicroPather/micropather.cpp

```diff
@@ -44,6 +45,9 @@ distribution.
 
 using namespace micropather;
 
+constexpr std::size_t Unset = std::numeric_limits<std::size_t>::max();

```

Comment:
Setting an `unsigned` variable to `-1` has well-defined and understood wrapping behavior. Declaring it as `constexpr auto Unset = std::size_t{-1};` is shorter, less repetitive of the type, and conveys the same information.

Reply:
I'm afraid Clang complains about that:
```cpp
constexpr auto Unset = std::size_t{-1};
```

> error: constant expression evaluates to -1 which cannot be narrowed to type 'std::size_t' (aka 'unsigned long') [-Wc++11-narrowing]
> constexpr auto Unset = std::size_t{-1};

Reply:
Try `constexpr auto Unset = std::size_t(-1u)` a non-brace-initialized version that doesn't enforce narrowing conversions.

Reply:
That works. Ran into this on another project, and someone resorted to using `numeric_limits` there.

Reply:
Goody. That said, a comment documenting the intended behavior is probably a good idea.

Reply:
Ehh, I don't know there is much to comment. I think most programmers understand wraparound.

Reply:
Actually, after examining the source code, there is an internal class-level maximum cache value `cacheCap` that limits the size of the path to some value "big enough" and is used everywhere `start` is used. Setting `Unset` to `cacheCap + 1` serves the same purpose and doesn't rely on wrap around.

Reply:
Yes, but `cacheCap` is a runtime value that can change. Much better to just use max and not worry about it.

Besides, this is for the code that was previously comparing to a hardcoded literal constant, which was previously set to `-1`. I'd rather not change the numerical value of the sentinel.

Reply:
Hm. Okay.